sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/aio/_azureappconfigurationproviderasync.py

```diff
@@ -391,7 +420,76 @@ async def refresh(self, **kwargs) -> None:
+                )
+                if update and not is_feature_flag:
+                    need_refresh = True
+                elif update and is_feature_flag:
+                    updated_feature_flags = True
+
+                sentinel_keys[(key, label)] = etag
+            # Need to only update once, no matter how many sentinels are updated
+            if need_refresh:
+                configuration_settings, sentinel_keys = self._load_configuration_settings(
+                    headers=headers, sentinel_keys=sentinel_keys, **kwargs
+                )
+            success = True
+            # Even if we don't need to refresh, we should reset the timer
+            timer.reset()
+        except (ServiceRequestError, ServiceResponseError, HttpResponseError) as e:
+            # If a call back is provided, we should call it, otherwise raise the error
+            if not self._on_refresh_error:
+                raise
+            self._on_refresh_error(e)
+        finally:
+            # If we get an error we should retry sooner than the next refresh interval
+            if not success:
+                timer.backoff()
+        return configuration_settings, sentinel_keys, need_refresh or updated_feature_flags
+
+    def _check_configuration_settings(self, key, label, etag, headers, configuration_settings, **kwargs) -> None:
+        is_feature_flag = False
+        try:
+            updated_config = self._client.get_configuration_setting(
+                key=key, label=label, etag=etag, match_condition=MatchConditions.IfModified, headers=headers, **kwargs
+            )
+            is_feature_flag = not isinstance(updated_config, FeatureFlagConfigurationSetting)
+            if updated_config is not None:
+                if not is_feature_flag:
+                    logging.debug("Refresh all triggered by key: %s label %s.", key, label)
+                elif is_feature_flag:
+                    configuration_settings[self._process_key_name(updated_config)] = updated_config.value
+                return updated_config.etag, True, is_feature_flag, configuration_settings
+        except HttpResponseError as e:
+            if e.status_code == 404:
+                if etag is not None and not is_feature_flag:
+                    # If the sentinel is not found, it means the key/label was deleted, so we should refresh
+                    logging.debug("Refresh all triggered by key: %s label %s.", key, label)
+                elif etag is not None and is_feature_flag:
+                    configuration_settings[key] = None
+                return None, True, is_feature_flag, configuration_settings
+            raise e
+        return etag, False, is_feature_flag, configuration_settings
+
     async def _load_all(self, **kwargs):
+        configuration_settings, sentinel_keys = await self._load_configuration_settings(**kwargs)
+        feature_flags, feature_flag_sentinel_keys = await self._load_feature_flags(**kwargs)
```

Comment:
This is a network call. Should it move below line 485 so we don't need to load feature flags if `feature_flag_enabled` is false?

Reply:
There is a check inside the method, but you are correct that we don't need to do this twice.